server/models/Document.ts

```diff
@@ -232,35 +232,50 @@ class Document extends ParanoidModel {
   // hooks
 
   @BeforeSave
-  static async updateInCollectionStructure(model: Document) {
-    if (!model.publishedAt || model.template) {
+  static async updateTitleInCollectionStructure(model: Document) {
+    // templates, drafts, and archived documents don't appear in the structure
+    // and so never need to be updated when the title changes
+    if (
+      model.archivedAt ||
+      model.template ||
+      !model.publishedAt ||
+      !model.changed("title")
```

Comment:
This is important, we're now only reading and updating the collection structure if the document title was edited.